helix-core/src/main/java/org/apache/helix/tools/ClusterSetup.java

```diff
@@ -176,6 +179,31 @@ public void close() {
     }
   }
 
+  public void addCluster(String clusterName, boolean overwritePrevious, CloudConfig cloudConfig)
+      throws HelixException {
+    if (cloudConfig != null) {
+      if (cloudConfig.isCloudEnabled()) {
+        String cloudID = cloudConfig.getCloudID();
+        String cloudProvider = cloudConfig.getCloudProvider();
+        List<String> cloudInfoSources = cloudConfig.getCloudInfoSources();
+        String cloudProcessorName = cloudConfig.getCloudInfoProcessorName();
+
+        // Validate the cloudConfig in the builder
+        // If it is invalid, do not even create a cluster
+        CloudConfig.Builder builder = new CloudConfig.Builder(clusterName, true,
+            CloudProvider.valueOf(cloudProvider), cloudID, cloudInfoSources, cloudProcessorName);
+
+        CloudConfig validatedCloudConfig = builder.build();
+        addCluster(clusterName, overwritePrevious);
```

Comment:
So as I mentioned above, if we have the validation method, we don't need to have 2 lines of addCluster, right?

Reply:
Addressed.